lib/mm_input.rb

```diff
@@ -1,16 +1,13 @@
-HELP = 'HELP'
-QUIT = 'QUIT'
-EXIT = 'EXIT'
-RULES = 'RULES'
+require_relative 'mm_outputs'
 
 class Input
-    attr_reader :user_input, :is_valid, :output_code, :restart_request, :show_help, :show_rules, :err_data
+    attr_reader :user_input, :is_valid, :output_code, :restart_request, :show_help, :show_rules, :err_data, :cmd_run
 
-    def initialize
+    def initialize(out = Outputs)
+        @out = Outputs
```

Comment:
the dependency injection here isn't actually doing anything because you're setting @out = Outputs anyway

Reply:
😳Good catch!